assets/admin/emails/email-preview-button/email-preview-button.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/compose';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { useState, createPortal } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { store as editorStore } from '@wordpress/editor';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * The email preview button component.
+ *
+ * @param {Object}  props
+ * @param {string}  props.previewLink    The preview link.
+ * @param {number}  props.postId         The post ID.
+ * @param {boolean} props.isSaveable     If the post is savable.
+ * @param {boolean} props.isAutosaveable If the post can be autosaved.
+ * @param {boolean} props.isLocked       If the post is locked.
+ * @param {boolean} props.isDraft        If the post is a draft.
+ * @param {Object}  props.autosave       The autosave action.
+ * @param {Object}  props.savePost       The save post action.
+ *
+ * @return {Object|null} A portal element or null if the container is not available.
+ */
+export const EmailPreviewButton = ( {
+	previewLink,
+	postId,
+	isSaveable,
+	isAutosaveable,
+	isLocked,
+	isDraft,
+	autosave,
+	savePost,
+} ) => {
+	const [ isSaving, setIsSaving ] = useState( false );
+	const container = document.querySelector(
```

Comment:
We should use useEffect or useLayoutEffect to make sure the component was rendered.

Reply:
Makes sense. Updated!